apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx

```diff
@@ -117,6 +123,40 @@ export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: Basic
         return PRIMARY_USERSTORE_PROPERTY_VALUES.RolenameJavaScriptRegEx;
     };
 
+    /**
+     * The following function validates role name against the user store regEx.
+     */
+    const validateRoleNamePattern = async (): Promise<string> => {
+        let userStoreRegEx = "";
+        if (userStore !== SharedUserStoreConstants.PRIMARY_USER_STORE) {
+            await SharedUserStoreUtils.getUserStoreRegEx(userStore,
+                SharedUserStoreConstants.USERSTORE_REGEX_PROPERTIES.RolenameRegEx)
+                .then((response) => {
+                    setRegExLoading(true);
+                    userStoreRegEx = response;
+                });
+        } else {
+            await SharedUserStoreUtils.getPrimaryUserStore().then((response) => {
+                setRegExLoading(true);
+                if (response && response.properties) {
+                    userStoreRegEx = response?.properties?.filter(property => {
+                        return property.name === "RolenameJavaScriptRegEx";
+                    })[ 0 ].value;
+                }
+            });
+        }
+
+        setRegExLoading(false);
+        return new Promise((resolve, reject) => {
```

Comment:
Can we add a new line above this?

Reply:
fixed with ac1667541af426a74f85a031d347125ff5038ad3